ui/src/app/base/components/FabricLink/FabricLink.tsx

```diff
@@ -0,0 +1,41 @@
+import { Spinner } from "@canonical/react-components";
+import { useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+
+import fabricSelectors from "app/store/fabric/selectors";
+import type { Fabric, FabricMeta } from "app/store/fabric/types";
+import { getFabricDisplay } from "app/store/fabric/utils";
+import type { RootState } from "app/store/root/types";
+import subnetsURLs from "app/subnets/urls";
+
+type Props = {
+  id?: Fabric[FabricMeta.PK] | null;
+};
+
+const FabricLink = ({ id }: Props): JSX.Element => {
+  const fabric = useSelector((state: RootState) =>
+    fabricSelectors.getById(state, id)
+  );
+  const fabricsLoading = useSelector(fabricSelectors.loading);
+  const fabricDisplay = getFabricDisplay(fabric);
+
+  if (fabricsLoading) {
+    // TODO: Put aria-label directly on Spinner component when issue is fixed.
+    // https://github.com/canonical-web-and-design/react-components/issues/651
+    return (
+      <span aria-label="Loading fabrics">
+        <Spinner />
+      </span>
+    );
+  }
+  if (!fabric) {
+    return <span aria-label="Fabric does not exist">{fabricDisplay}</span>;
```

Comment:
Is it a bit odd to have an aria-label but no visual content (same for all these components)?

Reply:
Yeah I suppose these could just return null instead